src/xenia/base/clock.cc

```diff
@@ -68,43 +132,58 @@ void Clock::set_guest_system_time_base(uint64_t time_base) {
 }
 
 uint64_t Clock::QueryGuestTickCount() {
-  UpdateGuestClock();
-  return guest_tick_count_;
+  auto guest_tick_count = UpdateGuestClock();
+  return guest_tick_count;
 }
 
 uint64_t Clock::QueryGuestSystemTime() {
-  UpdateGuestClock();
-  return guest_system_time_base_ + guest_time_filetime_;
-}
+  if (cvars::clock_no_scaling) {
+    return Clock::QueryHostSystemTime();
+  }
 
-uint32_t Clock::QueryGuestUptimeMillis() {
-  UpdateGuestClock();
-  uint64_t uptime_millis = guest_tick_count_ / (guest_tick_frequency_ / 1000);
-  uint32_t result = uint32_t(std::min(uptime_millis, uint64_t(UINT_MAX)));
-  return result;
+  auto guest_system_time_offset = QueryGuestSystemTimeOffset();
+  return guest_system_time_base_ + guest_system_time_offset;
 }
 
-void Clock::SetGuestTickCount(uint64_t tick_count) {
-  last_host_tick_count_ = Clock::QueryHostTickCount();
-  guest_tick_count_ = tick_count;
+uint32_t Clock::QueryGuestUptimeMillis() {
+  return static_cast<uint32_t>(
+      std::min(QueryGuestSystemTimeOffset() / 10000,
+               static_cast<uint64_t>(std::numeric_limits<uint32_t>::max())));
 }
 
 void Clock::SetGuestSystemTime(uint64_t system_time) {
-  last_host_tick_count_ = Clock::QueryHostTickCount();
-  guest_time_filetime_ = system_time - guest_system_time_base_;
+  if (cvars::clock_no_scaling) {
+    // Time is fixed to host time.
+    return;
+  }
+
+  // Query the filetime offset to calculate a new base time.
+  auto guest_system_time_offset = QueryGuestSystemTimeOffset();
+  guest_system_time_base_ = system_time - guest_system_time_offset;
 }
 
 uint32_t Clock::ScaleGuestDurationMillis(uint32_t guest_ms) {
-  if (guest_ms == UINT_MAX) {
-    return UINT_MAX;
+  if (cvars::clock_no_scaling) {
+    return guest_ms;
+  }
+
+  uint64_t constexpr max = std::numeric_limits<uint32_t>::max();
```

Comment:
It's `constexpr uint64_t` mostly according our clang-format, I think.

Reply:
Yes, in the morning that looks retarded.